### PR TITLE
Fix iotedge restart command with workload sockets (#6891)

### PR DESCRIPTION
**Pull request by gordonwang0**
Fix the `iotedge restart` command so that it sends a stop request followed by a start request. This is necessary so that workload sockets are properly notified of the restart.

### CODE REVIEW

File: edgelet/iotedge/src/client.rs [hunk 1]
```diff
@@ -146,10 +160,7 @@ impl ModuleRuntime for MgmtClient {
     async fn get(&self, _id: &str) -> anyhow::Result<(Self::Module, ModuleRuntimeState)> {
         unimplemented!()
     }
-    async fn start(&self, _id: &str) -> anyhow::Result<()> {
-        unimplemented!()
-    }
-    async fn stop(&self, _id: &str, _wait_before_kill: Option<Duration>) -> anyhow::Result<()> {
```
Comment (varunpuranik) [inline edgelet/iotedge/src/client.rs:152]: So, we are saying start/stop are now unimplemented?

Comment (gordonwang0) [reply edgelet/iotedge/src/client.rs:152]: No, they are now implemented. Restart is the one that is now unimplemented.

Comment (varunpuranik) [reply edgelet/iotedge/src/client.rs:152]: or is this just internal?

**Review by damonbarry — APPROVED** [review]
Varun says he talked to Gordon about the changes and believes they're ready to merge.

**Pull request merged**